src/github.com/matrix-org/go-neb/polling/polling.go

```diff
@@ -71,6 +72,15 @@ func pollLoop(service types.Service, ts int64) {
 		"service_id":   service.ServiceID(),
 		"service_type": service.ServiceType(),
 	})
+
+	defer func() {
+		if r := recover(); r != nil {
+			logger.WithField("panic", r).Errorf(
+				"pollLoop panicked!\n%s", debug.Stack(),
+			)
+		}
+	}()
```

Comment:
Might want to add metrics or something to track when things die.

Killing the loop might be the right thing here since it's likely that whatever cause us to panic will keep panic'ing

Maybe add some comments to explain what the intended behaviour is?